### ISSUE TITLE
9th parameter is transformed to number 8

### ISSUE THREAD

**Issue by mdarveau** [yortus/asyncawait#44]
I found out that if my function has exactly 9 parameters, the value of the 9th one is always replaced with the value 8.

Test case:
```
var async, await, test;

async = require('asyncawait/async');

await = require('asyncawait/await');

test = async(function(a, b, c, d, e, f, g, h, i) {
  console.log("a: " + a);
  console.log("b: " + b);
  console.log("c: " + c);
  console.log("d: " + d);
  console.log("e: " + e);
  console.log("f: " + f);
  console.log("g: " + g);
  console.log("h: " + h);
  console.log("i: " + i);
});

await(test('a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i'));
```

Output is:
```
a: a
b: b
c: c
d: d
e: e
f: f
g: g
h: h
i: 8
```

If I add one/remove one parameter from the function, everything is fine. If I remove the ```async``` part of the ```test``` method, everything is fine: the parameter value is unchanged.

What is going on here! :-)

Node version v4.1.2, asyncawait version 1.0.3.

**Comment by yortus** [comment]
Hi @mdarveau,

Thanks for reporting this. There was a bug [in the source](https://github.com/yortus/asyncawait/blob/master/src/async/makeAsyncFunc.ts#L152) where the nine argument was being clobbered. That scenario was clearly untested before you came along!

I've fixed the issue, added a test, and published v1.0.4 to npm. Can you please confirm that the issue is resolved?

**Comment by mdarveau** [comment]
Ah, javascript and variable scope...

Thank you for the quick response!

Manuel
